course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -0,0 +1,110 @@
+            $log.error('Error attempting to ' + config.method + ' ' + config.url +
+                ': ' + status + ' ' + statusText + ': ' + JSON.stringify(data));
+        };
+
+        dc.handleLookupResults = function(results) {
+            var courseResult = results[0];
+            var membersResult = results[1];
+
+            //check if the right data was obtained before storing it
+            if (courseResult.data.course_instance_id == dc.courseInstanceId) {
+                courseInstances.instances[courseResult.data.course_instance_id] = courseResult.data;
+                dc.courseInstance = dc.getFormattedCourseInstance(courseResult.data, membersResult.data)
+            } else {
+                $log.error(' CourseInstance record mismatch for id :'
+                    + dc.courseInstanceId + ',  fetched record for :' + courseResult.data.id);
+            }
+        };
+
+        dc.setCourseInstance = function (id) {
+
+            var course_url = djangoUrl.reverse(
+                'icommons_rest_api_proxy',
+                ['api/course/v2/course_instances/' + id + '/']);
+
+            var members_url = djangoUrl.reverse(
+                'icommons_rest_api_proxy',
+                ['api/course/v2/course_instances/'
+                + id + '/people/']);
+
+            var coursePromise = $http.get(course_url)
+                .error(dc.handleAjaxError);
+
+            var membersPromise = $http.get(members_url)
+                .error(dc.handleAjaxError);
+
+            $q.all([coursePromise, membersPromise])
+                .then(dc.handleLookupResults);
+        };
+
+        dc.stripQuotes = function(str){
+            // soem fields are coming over with quotes around them and those quotes
+            // are being displayed in the html.
+            // This strips off double quotes from the begining and ending of fields
+            return str ? str.trim().replace(new RegExp("^\"|\"$", "g"), "") : '';
+        };
+
+        dc.getFormattedCourseInstance = function (ci, members) {
+            // This is a helper function that formats the CourseInstance metadata
+            // and is combination of existing logic in
+            // Searchcontroller.courseInstanceToTable and Searchcontroller cell
+            // render functions.
+            courseInstance = {};
+            if (ci) {
+                courseInstance['title'] = dc.stripQuotes(ci.title);
+                courseInstance['school'] = ci.course ?
+                    ci.course.school_id.toUpperCase() : '';
+                courseInstance['term'] = ci.term ? ci.term.display_name : '';
+                courseInstance['year'] = ci.term ? ci.term.academic_year : '';
+                courseInstance['departments'] = ci.course.departments ? ci.course.departments : [];
+                courseInstance['course_groups'] = ci.course.course_groups ? ci.course.course_groups : [];
+                courseInstance['cid'] = ci.course_instance_id;
```

Comment:
sure, this was mostly copied from the people controller so it might also make sense to update it there as well. I'll put that in.